spectractor/simulation/simulator.py

```diff
@@ -291,6 +291,11 @@ def simulate_dispersion(self, D, shift_x, shift_y, r0):
             distance = distance[:-1]
         self.disperser.D = D
         lambdas = self.disperser.grating_pixel_to_lambda(distance, x0=new_x0, order=1)
+
+        distance -= adr_calib(lambdas,[self.dec,self.hour_angle,self.temperature,self.pressure,
```

Comment:
If you prefer, you can create an attribute self.adr_params = [self.dec,self.hour_angle,self.temperature,self.pressure,self.humidity,self.airmass,self.rotation_angle,self.xpixsize,self.ypixsize] to package this long list. It will avoid problems in the future in particular if one ady the arguments of adr_calib change.

Reply:
I think also that you should probably add adr line 65 (but check on your side too).

Reply:
Ok for self.adr_params
I don't think so, line 65, self.lambdas is an argument and this list comes from a spectrum.fits file which has already taken adr into account.

Reply:
Indeed but the lambdas are converted into pixels via the grating formula, I am not so sure the adr effect is still there after this operation even it was in the spectrum lambdas table.

Reply:
As in spectrum.py, grating_pixel_to_lambda and grating_lambda_to_pixel keep adr effect because there is a bijection between pixels and lambdas. For instance, lambda = 350 nm (350 pixels approximately) --> 360 nm with adr affect. When we use grating_lambda_to_pixel, 360 nm become around 360 pixels.

Reply:
Another way to phrase it: to go from the pixels to the lambdas in spectrum.py, you take the pixels from the PSF fit, you convert them in lambdas, you compute the adr pixel shifts, add theshifts to the pixels and compute the final lambdas. Now if we load the Spectrum lambdas and want  to go to the pixels, we have to do the exact same process but in reverse order. Am I right ? 
PS: in SpectrumSimulation, the only purpose to work on the lambdas is to add the shift of the order 0 xshift.